### DISCUSSION TITLE
Future of UWP/Windows Integration

### DISCUSSION THREAD

**Discussion by Coruscate5** [xamarin/Xamarin.Forms#12070] [General]
I've recently (over the course of 2 years) built a large-scale enterprise app that boasts all the good things about Xamarin Forms (mobile and desktop deployment, consistent UI).

Given the investment, how should I think about the future of its integration with UWP? It is difficult to tell how MSFT will steer the Windows UI ship from here (see SO's eternal confusion of web/WinRT/UWP).

Can I reasonably expect, with upgrades, that my Xamarin Forms app will continue to function on Windows for the foreseeable future? 
**Or maybe even better,** can we someday expect a Web-target deployment option from Xamarin Forms? (seems like generating a Webview from XAML would be tough, but not impossible)

Thanks!

**Comment by davidortinau** [comment]
@Coruscate5 Xamarin.Forms will continue run on the latest Windows has to offer, for the foreseeable and beyond. :) I would love to learn more about your application and your experiences building it. Email me and we can setup a chat: david.ortinau@microsoft.com

Right now the Windows implementation is UWP with some WinUI2 controls. We have done a spike of WinUI3 (see #11955) and will migrate to that in the .NET MAUI timeframe to align with WinUI3 GA (our teams sync weekly and will continue to coordinate release schedules). In terms of your existing app code using UWP ought to be a simple migration with little or no code change when WinUI3 becomes the API.

re: Web-target within Xamarin.Forms/.NET MAUI, we continue to listen for customer demand and learn how this might manifest. Today the [Uno Platform provides a path to take Xamarin.Forms to web](https://platform.uno/xamarin-forms/). I'd like to hear from you about this as well.

**Reply by Coruscate5** [reply]
This is great to hear! - was worried I should "pre-setup" a Web project using OOUI based [on some blog post on Telerik](https://www.telerik.com/blogs/xamarin-forms-on-the-web) to cover the unknown.

This is the first I've heard of the Uno platform - great to see web-targets in development. 

I will send an email to you from my corporate address re: chat. It's been a long road

**Comment by TheRexRegum** [comment]
I know this has nothing to do with this but looking for help if interested text me or email me at johnathanduvall.abu@gmail.com
443-358-7158. I need coders to build a state of the art social platform. could collaborate in near future more profits if we stand alone. Time is ticking I'm going to learn or the right people will surround me. Understand this ABU is coming. Join me now for everlasting prosperity for all.
